label_studio/ml/models.py

```diff
@@ -277,100 +331,13 @@ def predict_tasks(self, tasks):
             logger.debug(f'All tasks already have prediction from model version={self.model_version}')
             return model_version
         tasks_ser = TaskSimpleSerializer(tasks, many=True).data
-        ml_api_result = self.api.make_predictions(tasks_ser, self.project)
-        if ml_api_result.is_error:
-            logger.info(f'Prediction not created for project {self}: {ml_api_result.error_message}')
-            return
-
-        if not (isinstance(ml_api_result.response, dict) and 'results' in ml_api_result.response):
-            logger.info(f'ML backend returns an incorrect response, it should be a dict: {ml_api_result.response}')
-            return
-
-        responses = ml_api_result.response['results']
-
-        if len(responses) == 0:
-            logger.warning(f'ML backend returned empty prediction for project {self}')
-            return
-
-        # ML Backend doesn't support batch of tasks, do it one by one
-        elif len(responses) == 1 and len(tasks) != 1:
-            logger.warning(
-                f"'ML backend '{self.title}' doesn't support batch processing of tasks, "
-                f'switched to one-by-one task retrieval'
-            )
-            instances = [self.predict_one_task(task, model_version=model_version) for task in tasks]
-            return instances
-
-        # wrong result number
-        elif len(responses) != len(tasks_ser):
```

Comment:
this check len(responses) == 1 and len(tasks) != 1 this check may be useful - it helps to avoid user mistakes

Reply:
We have this check https://github.com/HumanSignal/label-studio/blob/fb-RND-97/label_studio/ml/models.py#L278

or are you talking about `len(responses) != len(tasks_ser)` ?

Reply:
I mean this : `len(responses) != len(tasks_ser)`
Imagine this: 

```
def predict(tasks): 
  results = []
  for task in tasks:
    try: 
      results.append(mode.predict(task))
    except:
      continue

  return results 
```

len(results) != len(task)

Reply:
The warning logs won't help too much, as in this situation, the task. IDs will be misaligned with predictions anyway. We should either completely fail here, or just ignore bad code on the ML backend side. I'd prefer the first approach instead of how it was implemented before.

Reply:
I believe we have to do lots of validations here and it would be one of them..